### PR TITLE
Relink some missing docs and remove unnecessary docs

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v6.1 (TiDB 6.1 versions)
- [x] v6.0 (TiDB 6.0 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/8724
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [x] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/dm-handle-performance-issues.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ summary: Learn about common performance issues that might exist in DM and how to
 
 This document introduces common performance issues that might exist in DM and how to deal with them.
 
-Before diagnosing an issue, you can refer to the [DM 2.0-GA Benchmark Report](/dm/dm-benchmark-v5.3.0.md).
+Before diagnosing an issue, you can refer to the [DM 2.0-GA Benchmark Report](https://github.com/pingcap/docs-dm/blob/release-5.3/en/dm-benchmark-v5.3.0.md).
```
Comment (shichun-0415) [inline dm/dm-handle-performance-issues.md:10]:
```suggestion
Before diagnosing an issue, you can refer to the [DM Benchmark Report](https://github.com/pingcap/docs-dm/blob/release-5.3/en/dm-benchmark-v5.3.0.md).
```

File: TOC.md [hunk 2]
```diff
@@ -368,6 +371,8 @@
       - [Glossary](/dm/dm-glossary.md)
     - Example
       - [Migrate Data Using DM](/dm/migrate-data-using-dm.md)
+      - [Create a data migration task](/dm/quick-start-create-task.md)
```
Comment (TomShawn) [inline TOC.md:374]:
```suggestion
      - [Create a Data Migration Task](/dm/quick-start-create-task.md)
```

**Review by TomShawn — APPROVED** [review]
Rest LGTM

**Review by sunzhaoyang — APPROVED** [review]
LGTM

**Pull request merged**